src/widgets/newsFeed/NewsFeed.tsx

```diff
@@ -0,0 +1,45 @@
+import styles from './news-feed.module.css'
+import Card from "~entities/card";
+import { Button } from "~shared/ui/button";
+import { Spinner } from "~shared/ui/spinner";
+import { articleApi } from "~entities/article";
+import { FullPageWrapper } from '~shared/ui/fullPageWrapper'
+
+type NewsFeedProps = {
+  query: articleApi.GlobalFeedQuery;
+}
+
+export function NewsFeed(props: NewsFeedProps) {
+  const { query } = props;
+
+  const {
+    data: articlesData,
+    status,
+    error,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage
+  } = articleApi.useGlobalInfinityArticles(query);
+
+  const pages = articlesData?.pages;
+  const items = pages?.flatMap((data) => data) || [];
+
+
+  if (status === 'loading') return <FullPageWrapper><Spinner/></FullPageWrapper>
+
+  if (status === 'error')
+    return <FullPageWrapper>{ error?.toString() }</FullPageWrapper>
+
+  return (
+      <div className={ styles.feedWrapper }>
+
+        { status === 'success' ? items.map((article: articleApi.IArticle) => (
+            <Card articleData={ article } key={ article.id }/>)) : 'No data' }
```

Comment:
which statuses can be in `status`? if only `loading`, `error` and `success` then you already checked `loading` and `error` above and it doesn't make sense to check `success` status here

better to check this logic 
```jsx
if (items.length === 0) {
  return <div className={ styles.feedWrapper }>No data</div>
}

return (
      <div className={ styles.feedWrapper }>
        {items.map((article: articleApi.IArticle) => (Card articleData={ article } key={ article.id }/>)}
        ...
```

Reply:
fixed